addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/handler/NikoHomeControlBridgeHandler.java

```diff
@@ -0,0 +1,238 @@
+     * Schedule future communication refresh.
+     *
+     * @param interval_config Time before refresh in minutes.
+     */
+    private void setupRefreshTimer(Integer refreshInterval) {
+
+        if (refreshTimer != null) {
+            refreshTimer.cancel(true);
+            refreshTimer = null;
+        }
+
+        if ((refreshInterval == null) || (refreshInterval == 0)) {
+            return;
+        }
+
+        // This timer will restart the bridge connection periodically
+        logger.debug("Niko Home Control: restart bridge connection every {} min", refreshInterval);
+        refreshTimer = scheduler.scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+
+                logger.debug("Niko Home Control: restart communication at scheduled time");
+
+                updateStatus(ThingStatus.OFFLINE);
+
+                try {
+                    for (int i = 0; i < 3; i++) {
+                        // try restarting max 3 times
+                        nhcComm.restartCommunication();
+                        if (nhcComm.communicationActive()) {
+                            break;
+                        }
+                        if (i == 2) {
+                            throw new IOException("Niko Home Control: communication socket error");
+                        }
+                    }
+                    addr = nhcComm.getAddr();
+                    updateProperties();
+
+                    updateStatus(ThingStatus.ONLINE);
+
+                } catch (IOException e) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                            "Niko Home Control: error restarting bridge connection");
+                }
+            }
+        }, refreshInterval, refreshInterval, TimeUnit.MINUTES);
+    }
+
+    /**
+     * Update bridge properties with properties returned from Niko Home Control Controller, so they can be made visible
+     * in PaperUI.
+     *
+     */
+    private void updateProperties() {
+
+        HashMap<String, String> properties = new HashMap<>();
+
+        properties.put("ipAddress", addr.getHostAddress());
+        properties.put("port", Integer.toString(port));
+        properties.put("softwareVersion", nhcComm.getSwVersion());
+        properties.put("apiVersion", nhcComm.getApi());
+        properties.put("language", nhcComm.getLanguage());
+        properties.put("currency", nhcComm.getCurrency());
+        properties.put("units", nhcComm.getUnits());
+        properties.put("tzOffset", nhcComm.getTz());
+        properties.put("dstOffset", nhcComm.getDst());
+        properties.put("configDate", nhcComm.getLastConfig());
+        properties.put("energyEraseDate", nhcComm.getLastEnergyErase());
+        properties.put("connectionStartDate", nhcComm.getTime());
+
+        thing.setProperties(properties);
+
+    }
+
+    @Override
+    public void dispose() {
+        nhcComm.stopCommunication();
```

Comment:
Please add a null check to the `nhcComm`, it might be called before the actual initialisation completed succesfully